app/schema/portal/cdo/dog-details.js

```diff
@@ -1,28 +1,41 @@
 const Joi = require('joi')
 const { UTCDate } = require('@date-fns/utc')
-const { isMatch, isValid, isFuture, addMonths } = require('date-fns')
+const { isValid, isFuture, addMonths, parse } = require('date-fns')
 const { dateComponentsToString } = require('../../../lib/date-helpers')
 
-const validateDate = (value, helpers) => {
-  const format = isMatch(value, 'yyyy-MM-dd') ||
-    isMatch(value, 'yyyy-M-d', { locale: 'enGB' }) ||
-    null
+const validDateFormats = [
+  'yyyy-MM-dd',
+  'yyyy-M-d'
+]
+
+const parseCdoIssueDate = (value) => {
+  for (const fmt of validDateFormats) {
+    const date = parse(value, fmt, new UTCDate())
+
+    if (isValid(date)) {
+      return date
+    }
+  }
 
-  if (!format) {
-    return helpers.message('CDO issue date must be a valid date')
+  return null
+}
+
+const validateDate = (value, helpers) => {
+  if (value === '--') {
+    return helpers.message('CDO issue date is required')
```

Comment:
Not sure if this message is defined in any AC but, to be consistent woth other error messages, it should really read 'Enter a CDO issue date'

Reply:
Good point - I'll correct now so it's consistent.